### PR TITLE
do not render env variables in configs

### PR DESCRIPTION
**Pull request by pjfanning**
see https://github.com/lightbend/config/pull/798

### CODE REVIEW

**Review by He-Pin — APPROVED** [review]
lgtm, do we need to change the code in spec too?

**Comment by mdedetrich** [comment]
Can we add a simple unit test for this?

**Comment by mdedetrich** [comment]
If it requires too much work we can skip it I guess, its just usually better to have a test for these kinds of things. Isn't `sys.env` mutable which might make things easier?

**Comment by pjfanning** [comment]
> If it requires too much work we can skip it I guess, its just usually better to have a test for these kinds of things. Isn't `sys.env` mutable which might make things easier?

I can play around with `sys.env` and see if that work. I also contacted you in Slack about another related issue that I don't want to make public yet.

**Comment by pjfanning** [comment]
@mdedetrich It looks like sys env is read only - likewise with `System.getenv()`. We might need to rely on the whether config jar is well tested. I have also added my own code to redact username.

**Comment by mdedetrich** [comment]
@pjfanning Okay, just ping me when the PR is ready. If its not possible to test then I would just go with the initial implementation (iirc it was simpler)

**Comment by pjfanning** [comment]
@mdedetrich I've kept the centralised method for rendering with redactions. My code does an additional redaction of the username and may in future do more. The toString of the config is very large and I wouldn't be surprised if we need to do more changes in future.

**Comment by mdedetrich** [comment]
Thanks ill check it out tomorrow

File: actor-tests/src/test/scala/org/apache/pekko/actor/ActorSystemSpec.scala [hunk 1]
```diff
@@ -392,6 +392,22 @@ class ActorSystemSpec extends PekkoSpec(ActorSystemSpec.config) with ImplicitSen
...
+          ConfigFactory
+            .parseString("""pekko.test.java.property.home = "${user.home}"""")
+            .withFallback(PekkoSpec.testConf))
+      val debugText = system.settings.toString
+      val username = System.getProperty("user.name")
+      val userHome = System.getProperty("user.home")
+      (debugText should not).include(username)
+      (debugText should not).include(userHome)
+      debugText should include("<username>")
+      shutdown(system)
```
Comment (mdedetrich) [inline actor-tests/src/test/scala/org/apache/pekko/actor/ActorSystemSpec.scala:409]: Minor nit, you can put this into a `try`/`finally` block so the actor system is shutdown even if an exception is thrown

**Review by mdedetrich — APPROVED** [review]
Minor nit but otherwise lgtm

**Pull request merged**